src/util/detect-bugs.js

```diff
@@ -1,23 +1,65 @@
-import TRANSITIONRUN_EVENT_LOOP_BUG from "./detect-bugs/transitionrun-loop.js";
-import UNREGISTERED_TRANSITION_BUG from "./detect-bugs/unregistered-transition.js";
+import detectTransitionRunLoopBug from "./detect-bugs/transitionrun-loop.js";
+import detectUnregisteredTransitionBug from "./detect-bugs/unregistered-transition.js";
 
-/**
- * Data structure for all detected bugs.
- * All bugs start off as true, and once their promises resolve, that is replaced with the actual value
- */
-export const bugs = {
-	TRANSITIONRUN_EVENT_LOOP: true,
-	UNREGISTERED_TRANSITION: true,
-};
+export default class Bugs {
+	/**
+	 * Whether the browser has the transition run loop bug
+	 * @type {boolean | null}
+	 * @private
+	 */
+	static #transitionRunLoop = null;
 
-TRANSITIONRUN_EVENT_LOOP_BUG.then(value => {
-	bugs.TRANSITIONRUN_EVENT_LOOP = value;
-});
+	/**
+	 * Whether the browser has the unregistered transition bug
+	 * @type {boolean | null}
+	 * @private
```

Comment:
I thought this could be inferred from the property itself. Do we really need to specify this explicitly?

Reply:
You are correct, thank you! Fixed in https://github.com/LeaVerou/style-observer/pull/111/commits/52152d32097d6cb201d72ea3110953430b5fe601.